src/components/select/filterable-select/filterable-select.component.js

```diff
@@ -308,11 +310,14 @@ const FilterableSelect = React.forwardRef(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [value, children]);
 
+    const onFilterChange = useStableCallback(onFilterChangeProp);
+    const isFirstRender = useRef(true);
     useEffect(() => {
-      if (onFilterChange) {
+      if (onFilterChange && !isFirstRender.current) {
         onFilterChange(filterText);
       }
-    }, [filterText, onFilterChange]);
+      isFirstRender.current = false;
```

Comment:
I'd rather keep the update of the `isFirstRender` ref in its own `useEffect` that's not dealing with anything else. It shouldn't change anything about either the component's current behaviour or how we might want to use `isFirstRender` in future, but it would keep each `useEffect` doing separate things (and ensure that in the unlikely event of us removing the `onFilterChange` functionality we don't accidentally forget to update the ref).